[kienvumrpm] iP

kienvumrpm opened the pull request:
# ToMo

> "Your mind is for having ideas, not holding them." - David Allen ([Source](https://dansilvestre.com/productivity-quotes/))

DukePro frees your mind of having to remember things you need to do. It's,

- text-based
- easy to learn
- ~FAST~ *SUPER* FAST to use

All you need to do is,

1. download it from [here](https://github.com/kienvumrpm/ip)
2. double-click it
3. add your tasks
4. let it manage your task for you 😉 

And it's **FREE**!

Features:

- [x] Adding tasks
- [x] Marking tasks
- [x] Filtering tasks
- [x] Tagging tasks
- [ ] Reminders (coming soon)  

If you are a Java programmer, you can use it to practice Java too. Here's the `main` method:
```
public class Main {
    public static void main(String[] args) {
        Application.launch(MainApp.class, args);
    }
}
```

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

aureliony committed:
build.gradle: Prevent generating a second JAR file

In build.gradle, the dependencies on distZip and/or distTar causes
the shadowJar task to generate a second JAR file for which the
mainClass.set("seedu.duke.Duke") does not take effect.
Hence, this additional JAR file cannot be run.
For this product, there is no need to generate a second JAR file
to begin with.

Let's remove this dependency from the build.gradle to prevent the
shadowJar task from generating the extra JAR file.

kienvumrpm committed:
Rename chatbot and add greeings

kienvumrpm committed:
Add echo to skeletal

kienvumrpm committed:
Close scanner

kienvumrpm committed:
feature add and list

kienvumrpm committed:
Add mark and unmark features

kienvumrpm committed:
add multiple types of tasks

kienvumrpm committed:
Automate testing

kienvumrpm committed:
refactor: move Task classes to outside

kienvumrpm committed:
Handle some invalid format exceptions

kienvumrpm committed:
Add delete command

kienvumrpm committed:
refactor by add OOPs

kienvumrpm committed:
add load and store method

kienvumrpm committed:
Merge pull request #1 from kienvumrpm/branch-Level-7

Branch level 7

kienvumrpm committed:
add DateTime features

kienvumrpm committed:
Merge pull request #2 from kienvumrpm/Branch-Level-8

add DateTime features

kienvumrpm committed:
Organize classes into packages

kienvumrpm committed:
Merge pull request #3 from kienvumrpm/Branch-A-Packages

Organize classes into packages

zzzlou commented on src/main/java/task/Task.java:
```diff
@@ -0,0 +1,25 @@
+
+    @Override
+    public String toString() {
+        String pref = isDone ? "[X] " : "[ ] ";
```
could more meaningful variable names be used? :)

zzzlou commented on src/main/java/tomo/ToMo.java:
Could JavaDoc be added to improve readability?

zzzlou commented on src/main/java/tomo/ToMo.java:
```diff
@@ -0,0 +1,104 @@
+    private TaskList tasks;
+    private Ui ui;
+
+    public ToMo(String fileName) {
```
Perhaps the name "ToMe" could be made clearer and more specific?

zzzlou commented on src/main/java/tomo/Parser.java:
Good job on barely making any common naming errors! 💯

kienvumrpm committed:
Merge remote-tracking branch 'upstream/add-gradle-support'

michaelyql commented on src/main/java/tomo/Parser.java:
```diff
@@ -0,0 +1,110 @@
+
+public class Parser {
+    public String[] parse(String cmd) throws ParserException {
+        if (cmd.split(" ").length == 0) return new String[]{};
```
Should there be curly braces to wrap this `if` statement?

michaelyql commented on src/main/java/task/Converter.java:
```diff
@@ -0,0 +1,18 @@
+import java.time.LocalDateTime;
+import java.time.format.DateTimeFormatter;
+
+public class Converter {
```
I like how you separated the date time conversion logic into a separate `Converter` class. Do you think we could improve it by giving it a more explicit name, to indicate that it's a converter for dates and time?

michaelyql commented on src/main/java/task/Event.java:
```diff
@@ -0,0 +1,23 @@
+
+    @Override
+    public String toString() {
+        return "[E]" + super.toString() + " (from: " + Converter.DateTimeToOutput(start) + " to: " + Converter.DateTimeToOutput(end) + ")";
```
nit: You can wrap a long string concatenation into a few lines if it helps with readability

michaelyql reviewed (commented):
LGTM. Just some nits to fix

kienvumrpm committed:
Add gradle

kienvumrpm committed:
Add junit

kienvumrpm committed:
Fix to compliant CodingStandard

kienvumrpm committed:
Add javadoc

kienvumrpm committed:
Add find support

kienvumrpm committed:
Merge pull request #4 from kienvumrpm/Branch-Level-9

Add find support

kienvumrpm committed:
Merge pull request #5 from kienvumrpm/Branch-A-JavaDoc

Add javadoc

kienvumrpm committed:
Merge pull request #6 from kienvumrpm/Branch-A-CodingStandard

Fix to compliant CodingStandard

kienvumrpm committed:
add command classes

kienvumrpm committed:
Merge pull request #8 from kienvumrpm/AddCommandClass

add command classes

kienvumrpm committed:
add checkstyle and fix style error

kienvumrpm committed:
Merge pull request #9 from kienvumrpm/Branch-A-CheckStyle

add checkstyle and fix style error

kienvumrpm committed:
Change behavious when the tasklist is empty

kienvumrpm committed:
add javafx

kienvumrpm committed:
add gui, remove ui, partially refactor tomo

kienvumrpm committed:
incorporate ToMo into gui, add handler on close request to replace bye command

kienvumrpm committed:
fix based on coding standard

kienvumrpm committed:
The storage is not ensured to be well formatted. Add assertions to ensure that each task ToDo, Deadline, Event is follow the specific format by convention. In general, assertions can address many problem related to format to prevent unexpected behaviour in later stages.

kienvumrpm committed:
Merge pull request #10 from kienvumrpm/Branch-A-Assertions

Add assertions

kienvumrpm committed:
Add assertion

The storage is not ensured to be well formatted. Add assertions to ensure that each task ToDo, Deadline, Event is follow the specific format by convention. In general, assertions can address many problem related to format to prevent unexpected behaviour in later stages.

kienvumrpm committed:
Add tagging

Implement a tag and untag function that allows users to achieve for better organization.

- Added tags features to Task class
- Added tag and untag command which allow users can quickly manage tag

This improves flexibily to add tag as well other issues to task.

kienvumrpm committed:
Merge pull request #11 from kienvumrpm/Branch-C-Tagging

Add tagging

kienvumrpm committed:
removes data and text-ui-test

kienvumrpm committed:
Changes images of User and Duke

kienvumrpm committed:
Revert "Changes images of User and Duke"

This reverts commit 2a1d8a8f258378ccf05f8c9fd5b407e736226480.

kienvumrpm committed:
Revert "removes data and text-ui-test"

This reverts commit 9ea15d9e6480d344f6282e0d3fb85a1da2bc973b.

kienvumrpm committed:
Remove older pictures

kienvumrpm committed:
Add JPG images to replace DaUser and DaDuke

kienvumrpm committed:
Remove redundant images

kienvumrpm committed:
Improve readibility of storage code and fix coding standard issue

kienvumrpm committed:
Fix storage issues

kienvumrpm committed:
Add Ui.png

kienvumrpm committed:
Modify README.md in root directory

kienvumrpm committed:
Update README.md

kienvumrpm committed:
Update Ui.png

kienvumrpm committed:
Update README.md

kienvumrpm committed:
Add JUnit back

kienvumrpm committed:
Fix space issues between tags from storage

kienvumrpm committed:
Update Ui.png
